local/owr_local_media_source.c

```diff
@@ -470,17 +518,38 @@ static GstElement *owr_local_media_source_request_source(OwrMediaSource *media_s
                 goto done;
             }
 
-            CREATE_ELEMENT(capsfilter, "capsfilter", "video-source-capsfilter");
-            source_caps = gst_caps_copy(caps);
-            source_structure = gst_caps_get_structure(source_caps, 0);
-            gst_structure_remove_field(source_structure, "format");
-
-            /* If possible try to limit the framerate at the source already */
-            if (gst_structure_get_fraction(source_structure, "framerate", &fps_n, &fps_d)) {
-                GstStructure *tmp = gst_structure_copy(source_structure);
-                gst_structure_remove_field(tmp, "framerate");
-                gst_caps_append_structure(source_caps, tmp);
+            /* First try to see if we can just get the format we want directly */
+
+            /* Framerate is handled at intervideosrc */
+            source_caps = gst_caps_new_empty();
+            gst_caps_foreach(caps, fix_video_caps_framerate, source_caps);
+
+            /* Now see what the device can really produce */
+            srcpad = gst_element_get_static_pad(source, "src");
+            gst_element_set_state(source, GST_STATE_READY);
+            device_caps = gst_pad_query_caps(srcpad, source_caps);
+
+            if (gst_caps_is_empty(device_caps)) {
+                /* Let's see if it works when we drop format constraints (which can be dealt with downsteram) */
+                GstCaps *tmp = source_caps;
+                source_caps = gst_caps_new_empty();
+                gst_caps_foreach(tmp, fix_video_caps_format, source_caps);
+                gst_caps_unref(tmp);
+
+                gst_caps_unref(device_caps);
+                device_caps = gst_pad_query_caps(srcpad, source_caps);
+
+                if (gst_caps_is_empty(device_caps)) {
+                    /* Accepting any format didn't work, we're going to hope that scaling fixes it */
+                    CREATE_ELEMENT(source_process, "videoscale", "video-source-scale");
+                    gst_bin_add(GST_BIN(source_pipeline), source_process);
```

Comment:
Why so complicated? Currently we only support raw caps anyway, so just always put a videoconvert and videoscale here :smile: More work is needed in these areas for compressed support anyway